autogpt_platform/backend/backend/blocks/code_executor.py

```diff
@@ -0,0 +1,182 @@
+from enum import Enum
+from typing import Literal
+
+from autogpt_libs.supabase_integration_credentials_store.types import APIKeyCredentials
+from e2b_code_interpreter import Sandbox
+from pydantic import SecretStr
+
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema
+from backend.data.model import CredentialsField, CredentialsMetaInput, SchemaField
+
+TEST_CREDENTIALS = APIKeyCredentials(
+    id="01234567-89ab-cdef-0123-456789abcdef",
+    provider="e2b_sandbox",
+    api_key=SecretStr("mock-e2b-api-key"),
+    title="Mock E2B API key",
+    expires_at=None,
+)
+TEST_CREDENTIALS_INPUT = {
+    "provider": TEST_CREDENTIALS.provider,
+    "id": TEST_CREDENTIALS.id,
+    "type": TEST_CREDENTIALS.type,
+    "title": TEST_CREDENTIALS.type,
+}
+
+
+class ProgrammingLanguage(Enum):
+    PYTHON = "python"
+    JAVASCRIPT = "js"
+    BASH = "bash"
+    R = "r"
+    JAVA = "java"
+
+
+class CodeExecutionBlock(Block):
+    # TODO : Add support to upload and download files
+    # Currently, You can customized the CPU and Memory, only by creating a pre customized sandbox template
+    class Input(BlockSchema):
+        credentials: CredentialsMetaInput[
+            Literal["e2b"], Literal["api_key"]
+        ] = CredentialsField(
+            provider="e2b",
+            supported_credential_types={"api_key"},
+            description="Enter your api key for the E2B Sandbox. You can get it in here - https://e2b.dev/docs",
+        )
+
+        # Todo : Option to run commond in background
+        commands: list[str] = SchemaField(
+            description="Run these commands in the sandbox before running the code, You can use `curl` or `git` to install your desired debian based package manager. pip and npm is pre installed\n Currently, none of these commands work in the background.",
```

Comment:
Break up into <88 length lines + small reword:
```suggestion
            description=(
                "Commands to set up the sandbox before running the code. You can "
                "use `curl` or `git` to install your desired Debian based "
                "package manager. `pip` and `npm` are pre-installed.\n\n"
                "Currently, these commands are not run in the background."
            ),
```